tar/read.c

```diff
@@ -67,6 +67,8 @@ __FBSDID("$FreeBSD: src/usr.bin/tar/read.c,v 1.40 2008/08/21 06:41:14 kientzle E
 
 #include "bsdtar.h"
 #include "err.h"
+#include "archive_platform.h"
```

Comment:
Try adding <io.h> to bsdtar_windows.h instead.

According to MSDN, _setmode() requires the <io.h> header and bsdtar source code includes bsdtar_windows.h (via bsdtar_platform.h) at the top of each source file when building on Windows.